src/libraries/System.Text.Json/src/System/Text/Json/Serialization/MemberAccessor.cs

```diff
@@ -9,17 +9,17 @@ namespace System.Text.Json
 {
     internal abstract class MemberAccessor
     {
-        public abstract JsonClassInfo.ConstructorDelegate CreateConstructor(Type classType);
+        public abstract JsonClassInfo.ConstructorDelegate? CreateConstructor(Type classType);
 
         public abstract Action<TProperty> CreateAddDelegate<TProperty>(MethodInfo addMethod, object target);
 
-        public abstract ImmutableCollectionCreator ImmutableCollectionCreateRange(Type constructingType, Type collectionType, Type elementType);
+        public abstract ImmutableCollectionCreator? ImmutableCollectionCreateRange(Type constructingType, Type collectionType, Type elementType);
 
-        public abstract ImmutableCollectionCreator ImmutableDictionaryCreateRange(Type constructingType, Type collectionType, Type elementType);
+        public abstract ImmutableCollectionCreator? ImmutableDictionaryCreateRange(Type constructingType, Type collectionType, Type elementType);
 
-        protected MethodInfo ImmutableCollectionCreateRangeMethod(Type constructingType, Type elementType)
+        protected MethodInfo? ImmutableCollectionCreateRangeMethod(Type constructingType, Type elementType)
         {
-            MethodInfo createRangeMethod = FindImmutableCreateRangeMethod(constructingType);
+            MethodInfo? createRangeMethod = FindImmutableCreateRangeMethod(constructingType);
 
             if (createRangeMethod == null)
```

Comment:
Similarly, this null check is unnecessary/unreachable. Change it to not return nullable.

Reply:
Yeah there is chain of null check and null return, callers of these methods also checking null and returning null ...

Reply:
Yep, those are also unreachable (at least from the tests perspective). I would remove them.

Reply:
removed